workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.ts

```diff
@@ -229,6 +234,23 @@ export class DynamicAuthorizationService {
     throw new Error('Not implemented');
   }
 
+  /**
+   * Get all identity permissions associated to a specific identity
+   * @param getIdentityPermissionsByIdentityRequest - {@link GetIdentityPermissionsByIdentityRequest}
+   *
+   * @returns - {@link GetIdentityPermissionsByIdentityResponse}
+   */
+  public async getIdentityPermissionsByIdentity(
+    getIdentityPermissionsByIdentityRequest: GetIdentityPermissionsByIdentityRequest
+  ): Promise<GetIdentityPermissionsByIdentityResponse> {
+    getIdentityPermissionsByIdentityRequest = GetIdentityPermissionsByIdentityRequestParser.parse(
```

Comment:
Im not sure its necessary to use zod to validate function params at the service level, but not a blocker

Reply:
This is probably way too deep of a comment but 🤷 haha. @htztomic and i were actually discussing this the other day. I think TypeScript's lack of runtime validation is a gaping hole in their "we're a typed language!" pitch and so its worth thinking about validation a little differently than you would with Java or something (where you only need logical validation and not typing validation). Validation should probably happen at all module boundaries because someone could be REALLY COOL and use an `as` or something to override the type system's static check and then break things at run time. Zod helps us close that gap a little bit by doing the runtime check via parsing.

If this were just a controller and service that were paired together and never being reused, then ya its probably overkill, but i think validation at the level the module with interface with other code is good to help us make sure our data is always what we expect it to be at run time. Thoughts?